run_tests.sh

```diff
@@ -17,14 +17,18 @@ sh $GOPATH/src/github.com/golang/dep/install.sh
 
 # run test suite
 dep ensure -v --vendor-only
-go test -v -timeout=10s ./tests/*
+go test -v -timeout=10s ./tests/* || exit 1
```

Comment:
Should you just `set -e` this file instead of instrumenting detectable failures?

Reply:
So I started with that approach, but the area you noted previously would cause things to fail every time. The loop works fine, but the `grep` results in a non-zero exit, so basically the script fails every time.

What I really wanted was just to make sure that when `go test` or `go lint` actually failed we would throw a real failure.

Reply:
Okay. I might throw some minutes at it if I ever trip over a future version of this; but for now, definitely an improvement. :+1:

Reply:
I'm going to be continuing to look in this area, so I suspect a few more things are likely to change here. Going to merge this down so that I can cause #38 to fail :)